WebDAV: update to v5

PTR-inc opened the pull request:
The currently used webdav package is v4 using an older axios dependency, giving a security warning in npm audit. The v5 version is an esm package, so converted the upload function accordingly. Now it's up to date again, also dependency-wise.

si458 commented:
you are using `import` instead of `require` 
wont this break existing installs?
also we dont specify `"type": "module"` in our package.json, so again wont this break meshcentral?

PTR-inc commented:
Done.

si458 commented on db.js:
```diff
@@ -3776,84 +3776,57 @@ module.exports.CreateDB = function (parent, func) {
-                    if (func) { func('Uploading using WebDAV to: ' + parent.config.settings.autobackup.webdav.url); }
-                });
-            }
+            const { stat } = await import ('fs/promises');
```
please can you change this to using `fs.statsync` instead?
https://nodejs.org/docs/latest-v16.x/api/fs.html#fsstatsyncpath-options

PTR-inc replied:
Changed it to using [pipeline](https://nodejs.org/en/learn/modules/how-to-use-streams#pipeline) instead of .pipe. No stat needed anymore. Cleaner, more resillient and futureproof. Tested it on 16.0.0 and latest 16, 22 & 23.

si458 commented:
this ok to merge still?

PTR-inc commented:
@si458 Certainly, if you are ok with the changes/async way. Just did a quick test against the current master and all fine.

PTR-inc commented:
@si458 Is there something keeping this on hold?

si458 commented:
no sorry just working my way through fixing a file transfer bug first then move onto this
its on my to-do to merge/verify for you 👍

si458 commented:
this still good to go?

PTR-inc commented:
Read about an issue somewhere else with trailing slashes and webdav, so checked the webdav rfc's and added a slashcheck for the foldername and moved it to the setup defaults sections.
